nextjs-legacy/lib/database.js

```diff
@@ -0,0 +1,53 @@
+import fs from 'node:fs/promises'
+import url from 'node:url'
+import path from 'node:path'
+
+let instance
+
+class Database {
+  #users = []
+
+  constructor(users) {
+    this.#users = users
+  }
+
+  allUsers() {
+    return this.#users
+  }
+
+  userById(id) {
+    return this.#users.find(user => user.id == id)
+  }
+
+  updateUserById(id, data) {
```

Comment:
this isn't really updating beyond the form. if i go back to /users it reverts back to the file

Reply:
If you step through the process you'll see that the data gets updated. I gave up trying to make the UI reflect it. I have no clue how to make it behave.

Reply:
it might be next.js caching, i know react has ways to force updates but not a bid deal